rust/private/rustc.bzl

```diff
@@ -584,6 +584,41 @@ def collect_inputs(
 
     return compile_inputs, out_dir, build_env_files, build_flags_files, linkstamp_outs, ambiguous_libs
 
+def _collect_ld_library_path(env, rust_toolchain, cc_toolchain):
+    """Construct the value of `LD_LIBRARY_PATH` based on the existing environment and registered toolchains.
+
+    Args:
+        env (dict): A dict of existing environment variables
+        rust_toolchain (rust_toolchain): The registered rust_toolchain
+        cc_toolchain (CcToolchainInfo): A cc toolchain provider to be used.
+
+    Returns:
+        str: The value of `LD_LIBRARY_PATH`.
+    """
+    sep = ";"
+    ld_paths = []
+
+    # Maintain existing values
+    if "LD_LIBRARY_PATH" in env:
+        ld_paths.extend(env["LD_LIBRARY_PATH"].split(sep))
```

Comment:
In addition to my impression that this env var (or equivalent linker flags) should be set by the C++ toolchain, maybe you could use https://docs.bazel.build/versions/main/command-line-reference.html#flag--action_env on the Bazel command line to achieve the same effect?

Reply:
I feel strongly that `--action_env` should not be used for `LD_LIBRARY_PATH`. This indicates a non hermetic aspect to your builds. That said though, this line attempts to handle cases where users set `--action_env`.

Reply:
Ok I think I understand, I hope. What I'm hearing is that if the user uses `--action_env`, we want to append to that, not replace it. So this thread is blocked by the resolution of the threads below.